CARDS-1574: Patient UI overhaul - Filling out surveys

marta- opened the pull request:
### Summary of changes:
* Bug fix: When there are no surveys to fill out, the ui skips to an empty review page after patient login
* Improvement: more meaningful message when there are no surveys to fill out
* Better wording on the screen listing incomplete surveys
* Shorter label for Review button
* Better wording on the Review screen
* Workflow fix: if there are only surveys with no interpretations for patients, adjust the wording so interpretations are not mentioned
* Better labels for the interpretation sections displayed to the patient user
* Improved vertical layout of review & summary screens
* Set a max with for the content on review/summary screens for better readability
### Testing:
* The implementation assumes that all necessary forms (_Patient information_, _Visit information_, all surveys) are created beforehand.
  * [testPatients.csv](https://github.com/data-team-uhn/cards/files/7922592/testPatients.csv) can be used to import **Patient information** forms:
    `curl -u admin:admin "localhost:8080/Forms/" -X POST -F ":data=@testPatients.csv" -F":questionnaire=/Questionnaires/Patient information" -F ":subjectType=/SubjectTypes/Patient"`
  * [testVisits.csv](https://github.com/data-team-uhn/cards/files/7922587/testVisits.csv) can be used to import Visit information and empty survey forms:
    * **Visit information**: `curl -u admin:admin "localhost:8080/Forms/" -X POST -F ":data=@testVisits.csv" -F":questionnaire=/Questionnaires/Visit information" -F ":subjectType=/SubjectTypes/Patient" -F ":subjectType=/SubjectTypes/Patient/Visit"`
    * **EQ5D**: `curl -u admin:admin "localhost:8080/Forms/" -X POST -F ":data=@testVisits.csv" -F":questionnaire=/Questionnaires/EQ5D" -F ":subjectType=/SubjectTypes/Patient" -F ":subjectType=/SubjectTypes/Patient/Visit"`
    * **PHQ9**: `curl -u admin:admin "localhost:8080/Forms/" -X POST -F ":data=@testVisits.csv" -F":questionnaire=/Questionnaires/PHQ9" -F ":subjectType=/SubjectTypes/Patient" -F ":subjectType=/SubjectTypes/Patient/Visit"`
    * **GAD7**: `curl -u admin:admin "localhost:8080/Forms/" -X POST -F ":data=@testVisits.csv" -F":questionnaire=/Questionnaires/GAD7" -F ":subjectType=/SubjectTypes/Patient" -F ":subjectType=/SubjectTypes/Patient/Visit"`
    * **AUDITC**: `curl -u admin:admin "localhost:8080/Forms/" -X POST -F ":data=@testVisits.csv" -F":questionnaire=/Questionnaires/AUDITC" -F ":subjectType=/SubjectTypes/Patient" -F ":subjectType=/SubjectTypes/Patient/Visit"`
    * **SC**: `curl -u admin:admin "localhost:8080/Forms/" -X POST -F ":data=@testVisits.csv" -F":questionnaire=/Questionnaires/SC" -F ":subjectType=/SubjectTypes/Patient" -F ":subjectType=/SubjectTypes/Patient/Visit"`
* Test at least the following scenarios:
  * There are no surveys to fill out (only create Patient information and Visit information forms)
  * There are surveys but they have already been submitted
  * There are only surveys without interpretation (e.g. EQ5D or SC)
  * The last survey is not paginated
  * The last survey is paginated
  * The patient forgets to answer some questions in the surveys
### To do / Out of scope:
* There are still too many requests to the server for the questionnaire set, survey forms, etc. This issue is not tackled by this PR.

marta- committed:
CARDS-1574: Patient UI overhaul
Better labels for the interpretation sections displayed to the patient user

marta- committed:
CARDS-1574: Patient UI overhaul
Bug fix: When there are no surveys to fill out, the ui skips to an empty review page after patient login
Improvement: more meaningful message when there are no surveys to fill out

marta- committed:
CARDS-1574: Patient UI overhaul
* Shorter label for Review button
* Better wording on the Review screen

marta- committed:
CARDS-1574: Patient UI overhaul
CARDS-1564: Better wording at the bottom of the patient's survey main screen

marta- committed:
CARDS-1574: Patient UI overhaul
Workflow fix: if there are only surveys with no interpretations for patients, adjust the wording so interpretations are not mentioned

marta- committed:
CARDS-1574: Patient UI overhaul
* Improved vertical layout of review & summary screens
* Set a max with for the content on review/summary screens for better readability

sdumitriu commented on proms-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/GAD7.xml:
```diff
@@ -274,7 +274,7 @@ Objectively determine initial symptoms severity and monitor symptom changes/effe
 			<property>
 				<name>text</name>
-				<value>GAD-7 Interpretation</value>
+				<value>General Anxiety Disorder-7</value>
```
Is the `-7` and `-9` in the next one really needed?

k-pace replied:
Yes, it is the specific name of the questionnaire (assuming you are referring to PHQ-9 and GAD-7)

sdumitriu reviewed (changes requested):
Regression:

- fill out all the surveys once, submit
- reopen the proms UI
- "Tell us about your symptoms", but nothing that I can interact with.

marta- committed:
CARDS-1574: Patient UI overhaul
Fix redression: once the surveys are complete and submitted, the UI displays "Tell us about your symptoms" after login, but nothing that the user can interact with.

marta- committed:
CARDS-1574: Patient UI overhaul
Improved spacing on review screen, simplified styling

marta- commented:
@sdumitriu all your comments have been addressed

sashaandjic reviewed (approved):
tested

k-pace commented:
Tested, looks great, two comments/things that I noticed (also in #testing for reference
- Upon completion of surveys as a patient, the page loads at the bottom rather than the top. This may be out of scope, apologies
- After I submitted my responses, if I log back in as the same patient I get prompted to review and (re)submit my responses. Is this expected?

sashaandjic commented:
> * After I submitted my responses, if I log back in as the same patient I get prompted to review and (re)submit my responses. Is this expected?

No, this should not be happening. I will check this scenario one more time.

sashaandjic commented:
I cannot reproduce the problem that @k-pace reported. Can someone else check this scenario?

k-pace commented:
> I cannot reproduce the problem that @k-pace reported. Can someone else check this scenario? 

Neither can I, so let's ignore the comment. I must have done something strange

marta- commented:
> Tested, looks great, two comments/things that I noticed (also in #testing for reference
> 
> * Upon completion of surveys as a patient, the page loads at the bottom rather than the top. This may be out of scope, apologies

If I understand correctly what is happening, this might be caused by the fact that the content is centered vertically, and as more content loads, anything that has already been displayed appears to be pushed up. Does that describe what you have seen? Is it bad enough to require fixing?

> * After I submitted my responses, if I log back in as the same patient I get prompted to review and (re)submit my responses. Is this expected?

I can replicate this issue only if the Visit information questionnaire is missing. Could that be the case in your instance?

k-pace commented:
> > Tested, looks great, two comments/things that I noticed (also in #testing for reference
> 
> > 
> 
> > * Upon completion of surveys as a patient, the page loads at the bottom rather than the top. This may be out of scope, apologies
> 
> 
> 
> If I understand correctly what is happening, this might be caused by the fact that the content is centered vertically, and as more content loads, anything that has already been displayed appears to be pushed up. Does that describe what you have seen? Is it bad enough to require fixing?

Nope

> 
> 
> 
> > * After I submitted my responses, if I log back in as the same patient I get prompted to review and (re)submit my responses. Is this expected?
> 
> 
> 
> I can replicate this issue only if the Visit information questionnaire is missing. Could that be the case in your instance?

I rewound and I think what happened is I double-imported the test patients, then deleted while I was logging in. I think this duplication might have been the culprit
